indico/modules/events/registration/models/tags.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from sqlalchemy.ext.declarative import declared_attr
+
+from indico.core.db import db
+
+
+class RegistrationTag(db.Model):
+    """Registration tag used to mark/filter registrations."""
+
+    __tablename__ = 'registration_tags'
+
+    @declared_attr
+    def __table_args__(cls):
+        return (db.Index('ix_uq_registration_tags_title_lower', db.func.lower(cls.title), unique=True),
+                {'schema': 'event_registration'})
+
+    #: The ID of the object
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    #: The ID of the event where this tag was created
+    event_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.events.id'),
+        index=False,
+        nullable=False
+    )
+    #: Tag name
+    title = db.Column(
+        db.String,
+        nullable=False
+    )
+    #: Tag color (hex)
+    color = db.Column(
+        db.String,
+        nullable=False
+    )
+    #: The Event where this tag was created
+    event = db.relationship(
+        'Event',
+        lazy=True,
+        backref=db.backref(
+            'registration_tags',
```

Comment:
```suggestion
            'registration_tags',
            cascade='all, delete-orphan',
```

While we do not hard-delete events, here we need the cascade setting since otherwise the `event_id` would get nulled (which isn't possible).